app/controllers/moderation_reports_controller.rb

```diff
@@ -0,0 +1,49 @@
+class ModerationReportsController < ApplicationController
+  respond_to :html, :xml, :json, :js
+  before_action :builder_only, only: [:new, :create]
+  before_action :moderator_only, only: [:index]
+
+  def new
+    check_privilege
+    @moderation_report = ModerationReport.new(moderation_report_params)
+    respond_with(@moderation_report)
+  end
+
+  def index
+    @moderation_reports = ModerationReport.paginated_search(params).includes(:creator, :model)
+    respond_with(@moderation_reports)
+  end
+
+  def create
+    check_privilege
+    @moderation_report = ModerationReport.create(moderation_report_params)
+    @moderation_report.create_forum_post!
```

Comment:
It's possible for `ModerationReport.create` to fail if the report fails validation. If this happens, `create_forum_post!` will create the forum post even though the moderation report didn't get created.

This can be fixed by calling `create_forum_post!` in an `after_create` hook on the model instead. This way if creating the report fails, then the after_create hook won't run.